src/libcore/array/mod.rs

```diff
@@ -202,6 +204,37 @@ where
     }
 }
 
+#[stable(feature = "array_from_iter_impl", since = "1.44.0")]
+impl<T, const N: usize> FromIterator<T> for [T; N] 
+where
+     [T; N]: LengthAtMost32,
+{
+    #[inline]
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
+        let mut iter = iter.into_iter();
+        let mut array: [MaybeUninit<T>; N] = MaybeUninit::uninit_array();
+        
+        for i in 0..N {
+            if let Some(value) = iter.next() {
```

Comment:
You still have a potential leak here if `iter.next()` itself panics. I would rewrite this logic using an `OnDrop` guard which does the cleanup on unwinding and if you successfully manage to collect, then `ManuallyDrop::new(drop_guard);`. Please also add a test case which exercises the path by panicking on e.g. the 3rd element.